platform/ios/src/MGLMapView.mm

```diff
@@ -763,26 +763,16 @@ - (UIViewController *)viewControllerForLayoutGuides
 
 - (void)updateConstraints
 {
-    // If we have a view controller reference and its automaticallyAdjustsScrollViewInsets
-    // is set to YES, use its view as the parent for constraints. -[MGLMapView adjustContentInset]
-    // already take top and bottom layout guides into account. If we don't have a reference, apply
-    // constraints against ourself to maintain placement of the subviews.
-    //
-    UIViewController *viewController = self.viewControllerForLayoutGuides;
-    BOOL useLayoutGuides = viewController.view && viewController.automaticallyAdjustsScrollViewInsets;
-    UIView *view = useLayoutGuides ? viewController.view : self;
-    
     // compass
     //
-    UIView *compassContainer = self.compassView.superview;
-    [view removeConstraints:self.compassViewConstraints];
+    [self removeConstraints:self.compassViewConstraints];
     [self.compassViewConstraints removeAllObjects];
 
     [self.compassViewConstraints addObject:
-     [NSLayoutConstraint constraintWithItem:compassContainer
+     [NSLayoutConstraint constraintWithItem:self.compassView
```

Comment:
> @incanus, do you remember why we stuck the compass view inside a container view in the first place?

Indeed! When you don't have it in a container, but constrain it to the corner using margins, it jiggles as the constraints attempt to keep up with the actual margins as they change. You'd think this would not be the case for a round image, but it is. So constraining the container keeps it solidly in place, changing only for device rotation and superview frame changes, while letting the round part rotate about its center, which is firmly constrained to the (from its perspective) unchanging container. 

So, if you get rid of the container, make sure the compass doesn't jiggle. It may have been a major iOS version-specific bug, so we should check all supported versions. 

https://github.com/mapbox/mi-sdk-legacy/commit/f7fdc9decaa6ad560969c5d2fb7ea7e3642a3159#diff-cc99abd5af5db8f48b362bcac3e8c307R514